core/src/test/java/io/opentelemetry/android/export/BufferDelegatingSpanExporter.kt

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.export
+
+import io.mockk.mockk
+import io.mockk.spyk
+import io.mockk.verify
+import io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.assertThat
+import io.opentelemetry.sdk.testing.exporter.InMemorySpanExporter
+import io.opentelemetry.sdk.trace.data.SpanData
+import org.junit.Test
+
+class BufferDelegatingSpanExporterTest {
+    @Test
+    fun `test setDelegate`() {
+        val bufferDelegatingSpanExporter = BufferDelegatingSpanExporter()
+        val spanExporter = InMemorySpanExporter.create()
+
+        val spanData: SpanData = mockk<SpanData>()
+        bufferDelegatingSpanExporter.export(listOf(spanData))
+        bufferDelegatingSpanExporter.setDelegate(spanExporter)
+
+        assertThat(spanExporter.finishedSpanItems)
+            .containsExactly(spanData)
+    }
+
+    @Test
+    fun `test buffer limit handling`() {
+        val bufferDelegatingSpanExporter = BufferDelegatingSpanExporter(10)
+        val spanExporter = InMemorySpanExporter.create()
+
+        for (i in 1..11) {
+            val spanData: SpanData = mockk<SpanData>()
+            bufferDelegatingSpanExporter.export(listOf(spanData))
+        }
+
+        bufferDelegatingSpanExporter.setDelegate(spanExporter)
+
+        assertThat(spanExporter.finishedSpanItems)
+            .hasSize(10)
+    }
+
+    @Test
+    fun `test flush with delegate`() {
+        val bufferDelegatingSpanExporter = BufferDelegatingSpanExporter()
+        val delegate = spyk<InMemorySpanExporter>()
+
+        val spanData: SpanData = mockk<SpanData>()
+        bufferDelegatingSpanExporter.export(listOf(spanData))
+
+        bufferDelegatingSpanExporter.setDelegate(delegate)
+
+        bufferDelegatingSpanExporter.flush()
+
+        verify { delegate.flush() }
```

Comment:
We should either verify that flush wasn't called before calling `flush` or verify call counts. Otherwise this will pass if `setDelegate` flushes

Reply:
good call out, done